main/test_utils.py

```diff
@@ -25,19 +28,23 @@ def assert_not_raises():
         pytest.fail(f"An exception was not raised: {traceback.format_exc()}")
 
 
-def assert_drf_json_equal(obj1, obj2):
+def assert_drf_json_equal(obj1, obj2, ignore_order=False):
     """
     Asserts that two objects are equal after a round trip through JSON serialization/deserialization.
     Particularly helpful when testing DRF serializers where you may get back OrderedDict and other such objects.
 
     Args:
         obj1 (object): the first object
         obj2 (object): the second object
+        ignore_order (bool): Boolean to ignore the order in the result
     """
     json_renderer = JSONRenderer()
     converted1 = json.loads(json_renderer.render(obj1))
     converted2 = json.loads(json_renderer.render(obj2))
-    assert converted1 == converted2
+    if ignore_order:
+        assert DeepDiff(converted1, converted2, ignore_order=ignore_order) == {}
+    else:
+        assert converted1 == converted2
```

Comment:
Could you explain why this change was needed/useful?

Reply:
Yup, I thought I addressed it, but I did not - 

The serializer (`CourseWithCourseRunsSerializer`) we use presently in `/courses` doesn't order the course runs, in theory you get them back in the same order, but I was getting mixed results.  The endpoints themselves are all ordered (I think, I might be wrong, I have not checked all), but the related objects are not. The order does not matter that I am aware of, so I added in this option to ignore order. 

This can be removed with changes to the API - the reason this worked prior and does not now is that we did not ever look at the course runs on the course endpoint - we would only add the top level model an endpoint returned, which meant the lack of ordering was not a problem prior as the course runs were not included on `/courses`.